connection.js

```diff
@@ -401,145 +436,368 @@ class Connection {
+            /* "blitz", "live", "correspondence" */
+            this.SpeedString = String(t.speed);
+
+            // before starting, sanity checks //
+            // sanity check : if not fischer, not byoyomi, not canadian, not simple, not absolute
+            if ((this.TimecontrolString !== "fischer") && (this.TimecontrolString !== "byoyomi") && (this.TimecontrolString !== "canadian") && (this.TimecontrolString !== "simple") && (this.TimecontrolString !== "absolute")) {
+                conn_log ("error, could not find allowed time control in " + t.time_control);
+                return { reject : true, msg: "error, could not find allowed time control in " + t.timecontrol};
+            }
+            // sanity check : if not "blitz" , not "live" , not "correspondence"
+            if ((this.SpeedString !== "blitz") && (this.SpeedString !== "live") && (this.SpeedString !== "correspondence")) {
+                conn_log ("error, could not find allowed game speed in " + t.speed);
+                return { reject : true, msg: "error, could not find allowed game speed in " + t.speed};
+            }
+            // -> then if sanity checks all pass :       
+
+            //////////// for blitz games : "blitz" //////////////////
+            // min
+            if ((config.minmaintimeblitz || config.minmaintimeblitzranked || config.minmaintimeblitzunranked) && (this.SpeedString === "blitz")) {
+                this.MinimumMaximumSentence = "Minimum ";
+                this.SpeedSentence = "for " + this.SpeedString + " ";
+                this.IncreaseDecreaseSentence = ", please increase ";
+                this.ConnBelowAboveSentence = " below ";
+                this.ConnSentence = user.username + " wanted " + this.TimecontrolString + this.ConnBelowAboveSentence + this.MinimumMaximumSentence + this.TimecontrolSentence ;
+                if (config.minmaintimeblitz && !config.minmaintimeblitzranked && !config.minmaintimeblitzunranked) {
+                    this.TimeNumber = config.minmaintimeblitz;
+                    this.TimeToString = timespanToDisplayString(config.minmaintimeblitz);
+                    this.RankedUnrankedGamesIsSentence = "games is ";
+                }
+                if (config.minmaintimeblitzranked && notification.ranked) {
+                    this.TimeNumber = config.minmaintimeblitzranked;
+                    this.TimeToString = timespanToDisplayString(config.minmaintimeblitzranked);
+                    this.RankedUnrankedGamesIsSentence = "ranked games is ";
+                }
+                if (config.minmaintimeblitzunranked && !notification.ranked) {
+                    this.TimeNumber = config.minmaintimeblitzunranked;
+                    this.TimeToString = timespanToDisplayString(config.minmaintimeblitzunranked);
+                    this.RankedUnrankedGamesIsSentence = "unranked games is ";
+                }
```

Comment:
3x duplicated code. I got the feeling this can be simplified.

Reply:
i can try to write a function, but for periodtime canadian it's not the same (it uses a different formula)

Reply:
oh for different formula, it might not be a function after all. Or possibly a smaller part can be made into a function?

Reply:
we can see the specific problem of canadian period time later, (only adds 3 more variables : 
- nc = n * stones;
- ncToString = timespanToDisplayString(nc);
- ncSentence = some small blablah

Reply:
but gonna try to UHMAEAT 3.0

it would indeed remove a lot of code maintainance